--- a/src/image.c	2014-12-03 20:58:33.000000000 +0900
+++ b/src/image.c	2015-08-07 13:15:47.595761688 +0900
@@ -75,6 +75,10 @@
 #include "jp2tools.h"
 #endif
 
+#ifndef GIFLIB_MAJOR
+#define GIFLIB_MAJOR 0
+#endif
+
 #define SRC_IMAGE_C
 #include "image.h"
 #include "cfgfile.h"
@@ -905,9 +909,15 @@
 	desc.size=insize;
 	desc.x.pos=0;
 
+#if GIFLIB_MAJOR < 5
 	if ((GifFile = DGifOpen((void*)&desc, &gif_mem_input)) == NULL) 
     		return( IMG_RET_ERR_UNKNOWN + IMG_RET_FLG_WHILE_DECOMP);//more possible reasons
 
+#else
+	if ((GifFile = DGifOpen((void*)&desc, &gif_mem_input, NULL)) == NULL) 
+    		return( IMG_RET_ERR_UNKNOWN + IMG_RET_FLG_WHILE_DECOMP);//more possible reasons
+#endif
+ 	bmp = new_raw_bitmap();
 	bmp = new_raw_bitmap();
 	*out = bmp;
 	bmp->width=GifFile->SWidth;
@@ -1083,7 +1093,11 @@
 	}
 	bmp->pal_bpp = bmp->bpp;
 	
+#if GIFLIB_MAJOR < 5
 	DGifCloseFile(GifFile);
+#else
+	DGifCloseFile(GifFile, NULL);
+#endif
 	return IMG_RET_OK;
 }
 
@@ -1738,7 +1752,7 @@
 		cmptparm->width = cwidth [cmptno];
 		cmptparm->height = cheight [cmptno];
 		cmptparm->prec = cbitlen [cmptno];
-		cmptparm->sgnd = false;	// we only generate unsigned components
+		cmptparm->sgnd = FALSE;	// we only generate unsigned components
 
 		cshiftR [cmptno] = 8 - cbitlen [cmptno]; // calculate the resampling shift for this component
 	}
